### ISSUE TITLE
CognitoCustomSMSSenderTestingLambdaHelperResources Create_Failed - Unknown Parameter AliasAttributes

### ISSUE THREAD

**Issue by RogeyRozza** [aws-samples/amazon-cognito-user-pool-development-and-testing-with-sms-redirected-to-email#1]
Hi Jian,

Thanks for sharing this!

I'm getting the following error when running the cloud formation json and yaml files (CustomSMSSenderWithExistingUserPool) 

`Received response status [FAILED] from custom resource. Message returned: Parameter validation failed: Unknown parameter in input: "AliasAttributes", must be one of: UserPoolId, Policies, LambdaConfig, AutoVerifiedAttributes, SmsVerificationMessage, EmailVerificationMessage, EmailVerificationSubject, VerificationMessageTemplate, SmsAuthenticationMessage, MfaConfiguration, DeviceConfiguration, EmailConfiguration, SmsConfiguration, UserPoolTags, AdminCreateUserConfig, UserPoolAddOns, AccountRecoverySetting`

I've specified the values for all the parameters as suggested as per your documentation. When running the stack it always fails when creating the CognitoCustomSMSSenderTestingLambdaHelperResources resource.

I don't see in the json/yaml files where the AliasAttributes parameter is specified so I am confused on where to debug and look.

Any chance you can help?

**Comment by jianwang000** [comment]
Hi RogeyRozza,

Thank you for reaching out! I was able to reproduce the issue. 

What happened is: when we use `CustomSMSSenderWithExistingUserPool` stack, the Lambda function`CognitoCustomSMSSenderTestingLambdaHelper` calls `describe_user_pool` and `update_user_pool` on behalf of us to 1) preserve existing configs and 2) add or remove CustomSMSSender Lambda configs for the existing user pool provided. Since the `"AliasAttributes"` cannot be update, we receive the exception when updating the use pool.

To quickly unblock your work, you can add `"AliasAttributes"` to `parameters_to_drop` list inside of the `CognitoCustomSMSSenderTestingLambdaHelper` Lambda function. The `parameters_to_drop` list contains parameters that are present in the `describe_user_pool` operation response but cannot be updated by the `update_user_pool` operation. Then your stack should be successfully deployed.

I will update the templates as well so that `"AliasAttributes"` will no longer be an issue for the user pools that contain this config. 

Please let us know if you have any questions. We are here to help!

Best regards,
Jian

**Comment by RogeyRozza** [comment]
Hi Jian,

Thanks for getting back so quickly.

Thankyou for the explanation, I've tried the suggest solution by adding `AliasAttributes ` to the `parameters_to_drop` and it works like a charm! Stack is successfully deployed.

All SMS's being sent through Cognito goes to the email address that's subscribed to the SNS topic.

Many thanks again for sharing this (you are the only one I came across who provides this solution), it does exactly what I want, to be able to test against user information in Cognito without having to send SMS's to the real user. 😊

Kind Regards,

Roger
